pkg/common/config/util.go

```diff
@@ -54,3 +57,38 @@ func GetAuthConfigFromSecret(secret *corev1.Secret) *client.KafkaAuthConfig {
 		},
 	}
 }
+
+// NumPartitions Gets The NumPartitions - First From Channel Spec And Then From ConfigMap-Provided Settings
+func NumPartitions(channel *kafkav1beta1.KafkaChannel, configuration *EventingKafkaConfig, logger *zap.SugaredLogger) int32 {
+	value := channel.Spec.NumPartitions
+	if value <= 0 && configuration != nil {
+		logger.Debug("Kafka Channel Spec 'NumPartitions' Not Specified - Using Default", zap.Int32("Value", configuration.Kafka.Topic.DefaultNumPartitions))
+		value = configuration.Kafka.Topic.DefaultNumPartitions
+	}
+	return value
+}
+
+// ReplicationFactor Gets The ReplicationFactor - First From Channel Spec And Then From ConfigMap-Provided Settings
+func ReplicationFactor(channel *kafkav1beta1.KafkaChannel, configuration *EventingKafkaConfig, logger *zap.SugaredLogger) int16 {
+	value := channel.Spec.ReplicationFactor
+	if value <= 0 && configuration != nil {
+		logger.Debug("Kafka Channel Spec 'ReplicationFactor' Not Specified - Using Default", zap.Int16("Value", configuration.Kafka.Topic.DefaultReplicationFactor))
+		value = configuration.Kafka.Topic.DefaultReplicationFactor
+	}
+	return value
+}
+
+// RetentionMillis Gets The RetentionMillis - First From Channel Spec And Then From ConfigMap-Provided Settings
+func RetentionMillis(channel *kafkav1beta1.KafkaChannel, configuration *EventingKafkaConfig, logger *zap.SugaredLogger) int64 {
+	//
+	// TODO - The eventing-contrib KafkaChannel CRD does not include RetentionMillis so we're
+	//        currently just using the default value specified in Controller Environment Variables.
+	//
```

Comment:
The `eventing-contrib` comment is out-of-date and should be reworded to reflect the history.  Essentially the current KafkaChannel spec doesn't have this and maybe we should look at adding it (was in old versions of distributed channel CRD.)

Reply:
I removed the RetentionMillis call and am now just using the Kafka.Topic.DefaultRetentionMillis value from the configmap directly where it is needed (consolidated reconcileTopic and distributed reconcileKafkaTopic).  I put the updated TODO comment in those locations so it will not be lost.